Library/src/Systems/HotSpotSequenceSystem/HotSpotSequenceSystem.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CSP/Systems/HotSpotSequence/HotSpotSequenceSystem.h"
+
+#include "CSP/Systems/HotSpotSequence/HotSpotGroup.h"
+#include "CSP/Systems/Sequence/SequenceSystem.h"
+#include "CSP/Systems/Spaces/SpaceSystem.h"
+
+#include <string>
+
+namespace csp::systems
+{
+HotSpotSequenceSystem::HotSpotSequenceSystem(csp::systems::SequenceSystem* SequenceSystem, csp::systems::SpaceSystem* SpaceSystem)
+{
+	this->sequenceSystem = SequenceSystem;
+	this->spaceSystem	 = SpaceSystem;
+}
+
+void HotSpotSequenceSystem::CreateHotspotGroup(csp::common::String GroupName,
+											   csp::common::Array<csp::common::String> HotspotIds,
+											   HotSpotGroupResultCallback Callback)
+{
+	auto spaceID			= spaceSystem->GetCurrentSpace().Id;
+	csp::common::String key = spaceID + ":" + GroupName;
+
+	auto CB = [Callback, key](SequenceResult result)
+	{
+		auto data = result.GetSequence();
+		HotSpotGroup group;
+		group.Items = data.Items;
+		group.Name	= data.Key;
+		HotSpotGroupResult returnValue(result.GetResultCode(), result.GetHttpResultCode());
+		returnValue.setHotSpotGroup(group);
```

Comment:
Could we pass the group into the result via the constructor and remove the need for a separate method?